spring-cloud-stream/src/test/java/org/springframework/cloud/stream/binder/DefaultSettingsTests.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsArrayContainingInOrder.arrayContaining;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.same;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.utils.MockBinderRegistryConfiguration;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Import;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class DefaultSettingsTests {
+
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testDefaultSettings() {
+		ConfigurableApplicationContext applicationContext = SpringApplication.run(TestFooChannels.class,
+				"--spring.cloud.stream.bindings.foo.destination=fooDest",
+				"--spring.cloud.stream.bindings.bar.destination=barDest",
+				"--spring.cloud.stream.bindings.baz.destination=bazDest",
+				"--spring.cloud.stream.bindings.qux.destination=quxDest",
+				"--spring.cloud.stream.bindings.foo.group=fooBarGroup",
+				"--spring.cloud.stream.bindings.bar.group=fooBarGroup",
+				"--spring.cloud.stream.consumerDefaults.concurrency=2",
+				"--spring.cloud.stream.producerDefaults.requiredGroups=quxbazReq1,quxbazReq2");
```

Comment:
Actually, I wanted to disable web/jmx altogether since that also helps with startup time.